core/auth/interfaces/authservice.go

```diff
@@ -0,0 +1,19 @@
+package interfaces
+
+import (
+	"context"
+	"net/url"
+	"flamingo.me/flamingo/v3/core/auth/domain"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	//Authservice - generic Authservice interface that should be used
+	Authservice interface {
+		Authenticate(ctx context.Context, returnURL *url.URL) (web.Result, error)
```

Comment:
As remainder to add web.Request in a list of arguments. We can omit to pass returnUrl - making of returnUrl is provided in security middleware, and it can be from header referrer or defined path in config (but, it doesn't hurt if we keep it).